### PR TITLE
Implements discord-json changes

### PR DESCRIPTION
**Pull request by Shadorc**
**Description:** Implements discord-json changes.

**Justification:** #894

Missing PRs:
- https://github.com/Discord4J/discord-json/pull/54
- https://github.com/Discord4J/discord-json/pull/57
- https://github.com/Discord4J/discord-json/pull/62

### CODE REVIEW

File: core/src/main/java/discord4j/core/event/domain/integration/IntegrationCreateEvent.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package discord4j.core.event.domain.integration;
```
Comment (quanticc) [inline core/src/main/java/discord4j/core/event/domain/integration/IntegrationCreateEvent.java:1]: Missing license header

File: core/src/main/java/discord4j/core/event/domain/integration/IntegrationDeleteEvent.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
+package discord4j.core.event.domain.integration;
```
Comment (quanticc) [inline core/src/main/java/discord4j/core/event/domain/integration/IntegrationDeleteEvent.java:1]: Missing license header

File: core/src/main/java/discord4j/core/event/domain/integration/IntegrationUpdateEvent.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
+package discord4j.core.event.domain.integration;
```
Comment (quanticc) [inline core/src/main/java/discord4j/core/event/domain/integration/IntegrationUpdateEvent.java:1]: Missing license header

File: core/src/main/java/discord4j/core/object/entity/Integration.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
+package discord4j.core.object.entity;
```
Comment (quanticc) [inline core/src/main/java/discord4j/core/object/entity/Integration.java:1]: Missing license header

**Review by quanticc — CHANGES_REQUESTED** [review]
Thanks a lot for keeping the project up to date with discord changes. I would like to merge this with rebase or fast-forward to keep commits local to a discord-json ticket and to make merging to 3.2.x simpler.

For that I have one request: Any chance we can get an update to each commit messages (via interactive rebase) to correctly refer to the ticket they cover? Now they point to a very old Discord4J issue instead of a discord-json issue.

**Comment by Shadorc** [comment]
Is it better now?

**Review by quanticc — COMMENTED** [review]
Awesome, thanks

**Review by quanticc — APPROVED** [review]

**Comment by darichey** [comment]
Should we take care of the couple TODOs before merging?

- I can't think of a use case for cached integrations
- I think the getters for Integration properties should be fairly straightforward

**Comment by Shadorc** [comment]
TODO have been adressed!

**Review by darichey — APPROVED** [review]

**Pull request merged**